### PR TITLE
debugging info for test timeouts

### PR DESCRIPTION
**Pull request by MHBauer**
Hoping for a couple more lines of output in failing and successful tests to try and get a feel for how long it takes for things to come up, and maybe tune that to more aggressively retry.

### CODE REVIEW

**Comment by MHBauer** [comment]
There is nothing specific to the apiserver in waitForApiserverUp. We could use it to point at the healthz endpoints of other servers.

**Comment by MHBauer** [comment]
https://travis-ci.org/kubernetes-incubator/service-catalog/builds/219902778
This is weird, because it takes the whole 2 minutes to timeout, but only prints three lines of 
`Waiting for : "https://localhost:7231"` 3 times, which it should print every 10 seconds.

**Comment by pmorie** [comment]
Yep, that's another puzzler that i whittled on and couldn't figure out

**Comment by pmorie** [comment]
@MHBauer IMO we shouldn't be afraid to make some changes to the checked in vendor dir and feed those changes back into the core.  The code that I saw in this area has almost no logging, which is of no help to anyone.

**Comment by MHBauer** [comment]
@pmorie I'd love to see more logging upstream, but I'm not sure where it needs to go, beyond, nearly everywhere.

**Comment by arschles** [comment]
LGTM on this - :100: on more logging. @MHBauer can you rebase? #695 was recently merged and I want to see if it fixes the `Waiting for : "https://localhost:7231"` error in the most recent CI run.

**Comment by MHBauer** [comment]
@sjenning while I'm in here, do you think you could help me hook up one of these queue jobbies so I can get the exponential backoff? 

I think that would be a good thing to understand the queue usage, and maybe form a pattern around it for reuse on all of the other "client calls". Maybe resulting later in an upstream integration that can go back to client-go.

**Comment by MHBauer** [comment]
oh derp. I should look at `pkg/util/wait.go#ExponentialBackoff`.

**Comment by MHBauer** [comment]
@arschles @simonleung8 re-review. Changed the loop to use util/wait from upstream.

File: test/integration/framework.go [hunk 1]
```diff
@@ -107,23 +109,27 @@ func getFreshApiserverAndClient(t *testing.T, storageTypeStr string) (servicecat
...
-		default:
-			glog.Infof("Waiting for : %#v", insecureAddr)
-			_, err := http.Get(insecureAddr)
-			if err == nil {
-				return nil
+func waitForApiserverUp(serverURL string, stopCh <-chan struct{}) error {
+	startWaiting := time.Now()
+	tries := 0
+	// roughly 30 seconds total
+	return wait.ExponentialBackoff(wait.Backoff{1 * time.Second, 2, 0, 5},
```
Comment (arschles) [inline test/integration/framework.go]: can you `PollImmediate` here? It makes more sense for a test that we proceed as soon as possible, at the expense of some CPU and allocations

**Comment by arschles** [comment]
@MHBauer I was fine with the loop, but if you stick with this, I left a comment: https://github.com/kubernetes-incubator/service-catalog/pull/671/files#r111506593

File: test/integration/framework.go (hunk 1 above)
Comment (MHBauer) [inline test/integration/framework.go]: Ya, sure. It takes up to a couple seconds to start up usually. ~3-7 in my runs.

**Comment by MHBauer** [comment]
travis seems to be faster on this run. 2-4s to start. look for "tries and duration"

**Pull request merged**